lib/gateways/sessions.rb

```diff
@@ -3,15 +3,18 @@ class Sessions
     MAXIMUM_RESULTS_COUNT = 500
 
     def initialize(ips:)
-      @ips = ips
+      @organisation_ips = ips
     end
 
-    def search(username: nil, ip: nil)
-      query = username.present? ? { username: username } : { siteIp: ip }
+    def search(username: nil, ips: nil)
+      results = Session
+        .where(siteIP: organisation_ips)
+        .where('start >= ?', 2.weeks.ago)
+        .order(start: :desc)
+        .limit(MAXIMUM_RESULTS_COUNT)
```

Comment:
We're running 2 queries?
Does this above load the results into memory?

Reply:
If this is using arel under the hood, it's lazy and won't execute the query 'til the `.map` call. I can double-check that, though.